__tests__/components/NetworkSelector.test.tsx

```diff
@@ -5,16 +5,30 @@ import * as nextRouter from 'next/router';
 import { NetworkSelector } from 'components/NetworkSelector';
 
 jest.spyOn(nextRouter, 'useRouter');
+const originalLocation = global.location;
 
 const mockedUseRouter = nextRouter.useRouter as jest.MockedFunction<
   typeof nextRouter.useRouter
 >;
-const push = jest.fn();
 mockedUseRouter.mockImplementation(
-  () => ({ push, route: '/network/rinkeby/loans/create' } as any),
+  () => ({ route: '/network/rinkeby/loans/create' } as any),
 );
 
+const mockAssign = jest.fn();
+
 describe('Select', () => {
+  beforeAll(() => {
+    Object.defineProperty(window, 'location', {
```

Comment:
this is a bit of an awkward mock due to the way JSDom works. Can't just spy on the existing function